lib/src/analyzer/rules/use_alternative_class/use_alternative_class_visitor.dart

```diff
@@ -0,0 +1,61 @@
+import 'dart:collection';
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer_plugin/protocol/protocol_generated.dart';
+
+import '../../rule/rule_visitor.dart';
+import '../../rule/utils.dart';
+import 'use_alternative_class.dart';
+
+class UseAlternativeClassVisitor extends RecursiveRuleVisitor {
+  UseAlternativeClassVisitor({
+    required super.rule,
+    required super.result,
+    required super.config,
+  }) {
+    final allowedClasses = config.options[allowedClassesKey] as List<Object>?;
+
+    _alternatives =
+        (config.options[alternativeKey] as UnmodifiableMapView<String, Object?>?)
+                ?.entries.where((e) => allowedClasses == null || !allowedClasses.contains(e.key))
+                .map((e) => MapEntry(e.key, e.value! as String)).toList() ?? [];
```

Comment:
Avoid using `as`. Prefer `is` instead. It is safer and will not crash the analyzer.

```suggestion
    _alternatives = {};
    final alternatives = config.options[alternativeKey]

    if (alternatives is Map<String, Object?>) {
      for (final alternative in alternatives) {
        final replacement = alternative.value;
        if (replacement is String) {
          _alternatives.add(alternative.key, replacement);
        }
      }
    }
```

Reply:
made changes to use `is`. thank you!